Composer/packages/client/src/components/TestController/TestController.tsx

```diff
@@ -114,39 +119,75 @@ export const TestController: React.FC = () => {
     setCalloutVisible(true);
   }
 
-  async function handlePublishLuis(luisConfig) {
+  async function handlePublish(config) {
     setBotStatus(BotStatus.publishing);
     dismissDialog();
-    await setSettings(projectId, { ...settings, luis: luisConfig });
-    await publishLuis(luisConfig, projectId);
+    // save the settings change to store and persist to server
+    const newValue = config;
+    const subscriptionKey = newValue.subscriptionKey;
+    const qnaRegion = newValue.qnaRegion;
+    delete newValue.subscriptionKey;
+    delete newValue.qnaRegion;
```

Comment:
The processing of `config` is splited into two parts in `handleStart()`
![image](https://user-images.githubusercontent.com/8528761/90357583-0c033e00-e086-11ea-8013-4b6978386c59.png)

`handleStart()` serves `<ErrorCallout />` and `handlePublish()` serves both `handleStart()` and `<PublishDialog>`. It's a fragile config processing chain.

Try to extract the config processing logic.